### PR TITLE
Introduce db entity for model revision

### LINKED ISSUE #4672: Create table and repo for model revisions
Create the `model_revisions` table and relative repo, as per design.

Image

Fields:
- `id`: unique identifier
- `project_id`: foreign key linking to the project that owns the model
- `architecture`: identifier of the model architecture (same as Geti manifest id)
- `parent_revision`: reference to the parent model revision (if any) that was fine-tuned to obtain this revision
- `training_status`: could be "not_started", "in_progress", "failed", "successful"
- `training_configuration`: object containing the full training configuration, hyperparameters, ...
- `training_dataset_id`: reference to the dataset revision used for training
- `label_schema_revision`: label schema used to train the model, encoded as JSON
- `training_started_at`: timestamp when the training job started
- `training_finished_at`: timestamp when the training job finished
- `files_deleted`: whether the binary files of this model are still available or not

### PR DESCRIPTION
**Pull request by itallix**
### Summary

This PR introduces a new database entity - `ModelRevisionDB`, which will be used to represent metadata for tuned models. Additionally, it removes obsolete code related to the previous Bring Your Own Model functionality.

- [x] Added `ModelRevisionDB` entity and updated references.
- [x] Removed outdated BYOM-related code and references.

### How to test

1. Rename the model artifact files in the `data/models` directory to include the revision ID (`977eeb18-eaac-449d-bc80-e340fbe052ad`) in their names.
2. Remove `data/geti_tune.db` if exists
3. Run the following command:

```bash
SEED_DB=true ./run.sh
```

4. The script should execute without errors and inference results appear in `data/output` directory.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/open-edge-platform/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/open-edge-platform/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/open-edge-platform/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by jpggvilaca — DISMISSED** [review]

File: backend/app/repositories/model_revision_repo.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class ModelRevisionRepository(BaseRepository[ModelRevisionDB]):
+    """Repository for model-related database operations."""
+
+    def __init__(self, db: Session):
+        super().__init__(db, ModelRevisionDB)
+
+    def __get_active_pipeline(self) -> PipelineDB | None:
+        """Get the active pipeline from database."""
+        return self.db.query(PipelineDB).filter(PipelineDB.is_running).first()
```
Comment (warrkan) [inline backend/app/repositories/model_revision_repo.py]: Shouldn't it be a part of `PipelineRepository`?

File: backend/tests/conftest.py [hunk 2]
```diff
@@ -33,7 +34,10 @@ def fxt_mqtt_sink() -> MqttSinkConfig:
 @pytest.fixture
 def fxt_model() -> Model:
     """Sample model data."""
-    return Model(name="YOLO-X for Vehicle Detection", format=ModelFormat.OPENVINO)
+    return Model(
```
Comment (warrkan) [inline backend/tests/conftest.py:37]: Don't we need format anymore?

Comment (itallix) [reply backend/tests/conftest.py:37]: Yes, we don't - [these requirements](https://github.com/open-edge-platform/training_extensions/blob/develop/docs/geti_tune/pipeline.md#models) took precedence to favor a single format:

> A pipeline performs inference using models in OpenVINO IR format (.xml + .bin), converted after fine-tuning. The supported task types include image classification, object detection and instance segmentation.

File: backend/app/repositories/model_revision_repo.py (hunk 1 above)
Comment (itallix) [reply backend/app/repositories/model_revision_repo.py]: Good point! I rewrote the method to use a single SQL query.

**Review by warrkan — APPROVED** [review]

**Review by jpggvilaca — APPROVED** [review]
